lib/config/common.ts

```diff
@@ -198,19 +197,19 @@ export interface PackageRule
   extends RenovateSharedConfig,
     UpdateConfig,
     Record<string, any> {
-  paths?: string[];
-  languages?: string[];
-  baseBranchList?: string[];
-  datasources?: string[];
-  depTypeList?: string[];
-  packageNames?: string[];
-  packagePatterns?: string[];
+  matchPaths?: string[];
+  matchLanguages?: string[];
+  matchBaseBranches?: string[];
+  matchManagers?: string | string[];
+  matchDatasources?: string[];
+  matchDepTypes?: string[];
+  matchPackageNames?: string[];
+  matchPackagePatterns?: string[];
   excludePackageNames?: string[];
   excludePackagePatterns?: string[];
```

Comment:
Maybe we can remove those `exclude**` options and simply use a negation pattern, like prefix with `!` ?

Reply:
Should be a separate PR then, probably first. Negation should suit package names ok, not sure about patterns though as presumably it's allowed to start a regex with `!`